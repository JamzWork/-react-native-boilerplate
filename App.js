/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow strict-local
 */

import React, { Component} from 'react';
 
 
import AppNavigator from './App/Navigators/AppNavigator'
import { Provider } from "react-redux";
import store from './App/Store/Store';

class App extends Component {
  constructor(props){
    super(props)
  }

  componentDidMount(){
    
  }

  render(){
    return( 
      <Provider store={store}>
        <AppNavigator/> 
      </Provider>
    )
  }
}
 
export default App;
