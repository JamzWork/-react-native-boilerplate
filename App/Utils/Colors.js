export default{
    white:'#ffffff',
    black:'#000000',
    AppTheme:'#002366'
}