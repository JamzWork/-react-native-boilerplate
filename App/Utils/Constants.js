export const TextSize  = 15;
export const AppHeaderTextSize = 18;