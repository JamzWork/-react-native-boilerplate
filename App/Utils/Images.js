export default {
    Splash_Icon:  require('../Assets/Images/icon_react.png'),
}