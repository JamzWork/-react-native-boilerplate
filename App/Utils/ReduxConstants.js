export const SPINNER_ON = "Spinner_on";
export const SPINNER_OFF = "Spinner_off";