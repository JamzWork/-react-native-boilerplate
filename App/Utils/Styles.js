import * as Constants from './Constants'; 
import Colors from "./Colors"; 

export default {
    CommonStyles: {
        full_flex: { flex: 1 },
        centerStyles: { justifyContent: 'center', alignItems: 'center',alignSelf:'center'},
    },
    SplashScreen:{
        ImageStyle: {alignSelf:'center', width:200, height:200},
        Text:{fontSize:22}
    },
    HomeScreen:{

    },
    Header:{
        appHeaderText: {  color:Colors.white ,flex:1, textAlign:'center' ,  fontSize: Constants.AppHeaderTextSize,  fontWeight: 'bold' },
        headerSize: {backgroundColor: Colors.AppTheme ,width:"100%",flexDirection: 'row' ,alignItems:'center', height:50}
    }
}