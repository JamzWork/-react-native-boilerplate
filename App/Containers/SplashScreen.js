import React, { Component } from 'react'
import { Text, View, Image, Alert } from 'react-native' 
import mStyles from '../Utils/Styles'
import Images from '../Utils/Images'

class SplashScreen extends Component{

    constructor(props){
        super(props);
    }

    render(){ 
        return( 
            <View style={[mStyles.CommonStyles.centerStyles,mStyles.CommonStyles.full_flex]}>
                <View style={[ mStyles.full_flex]}>
                    <Image source={Images.Splash_Icon} style={mStyles.SplashScreen.ImageStyle} resizeMode={'contain'}/>
                    <Text style={mStyles.SplashScreen.Text}>React Native Boilerplate</Text>
                </View> 
            </View>
        )
    }

    componentDidMount(){
        // 2 Seconds for splash screen
        setTimeout(()=>{  
            this.props.navigation.navigate('HomeScreen');
        }, 2000)
    }

    componentWillUnmount(){  
    }
}
  
 

export default SplashScreen