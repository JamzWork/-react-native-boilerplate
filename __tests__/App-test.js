/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';
import SplashScreen from '../App/Containers/SplashScreen'

import configureStore from 'redux-mock-store';
 
import renderer from 'react-test-renderer'; 
import { SPINNER_ON } from  '../App/Utils/ReduxConstants'
it('renders correctly', () => {
  renderer.create(<App />);
});

it('renders correctly across screens', () => {
  const tree = renderer.create(<SplashScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
 
const middlewares = []
const mockStore = configureStore(middlewares)

// You would import the action from your codebase in a real scenario
const SpinnerON = () => ({ type: SPINNER_ON })

it('should dispatch action', () => {
 
  const initialState = {}
  const store = mockStore(initialState)

  // Dispatch the action
  store.dispatch(SpinnerON())

  // Test if your store dispatched the expected actions
  const actions = store.getActions()
  const expectedPayload = { type: SPINNER_ON }
  expect(actions).toEqual([expectedPayload])
})