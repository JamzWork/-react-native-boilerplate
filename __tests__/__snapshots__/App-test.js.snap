// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly across screens 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "justifyContent": "center",
      },
      Object {
        "flex": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        undefined,
      ]
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../App/Assets/Images/icon_react.png",
        }
      }
      style={
        Object {
          "alignSelf": "center",
          "height": 200,
          "width": 200,
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 22,
        }
      }
    >
      React Native Boilerplate
    </Text>
  </View>
</View>
`;
